glitch/analysis/security.py

```diff
@@ -1,15 +1,1107 @@
+                            errors.append(Error('sec_key_management', element, file, repr(element)))
+                            break
+
+                if (elem_name == "rotation_period" and au_type == "resource.google_kms_crypto_key"):
+                    expr1 = r'\d+\.\d{0,9}s'
+                    expr2 = r'\d+s'
+                    if (re.search(expr1, elem_value) or re.search(expr2, elem_value)):
+                        if (int(elem_value.split("s")[0]) > 7776000):
+                            errors.append(Error('sec_key_management', element, file, repr(element)))
+                    else:
+                        errors.append(Error('sec_key_management', element, file, repr(element)))
+                elif (elem_name == "kms_master_key_id" and ((au_type == "resource.aws_sqs_queue"
+                    and elem_value == "alias/aws/sqs") or  (au_type == "resource.aws_sns_queue"
+                    and elem_value == "alias/aws/sns"))):
+                    errors.append(Error('sec_key_management', element, file, repr(element)))
+            return errors
+
+    class TerraformNetworkSecurityRules(TerraformSmellChecker):
+        def check(self, element, file: str, code, elem_name: str, elem_value: str = "", au_type = None, parent_name = ""):
+            errors = []
+            if isinstance(element, AtomicUnit):
+                if (element.type == "resource.azurerm_network_security_rule"):
+                    access = self.check_required_attribute(element.attributes, [""], "access")
+                    if (access and access.value.lower() == "allow"):
+                        protocol = self.check_required_attribute(element.attributes, [""], "protocol")
+                        if (protocol and protocol.value.lower() == "udp"):
+                            errors.append(Error('sec_network_security_rules', access, file, repr(access)))
+                        elif (protocol and protocol.value.lower() == "tcp"):
+                            dest_port_range = self.check_required_attribute(element.attributes, [""], "destination_port_range")
+                            dest_port_ranges = self.check_required_attribute(element.attributes, [""], "destination_port_ranges[0]")
+                            port = False
+                            if (dest_port_range and dest_port_range.value.lower() in ["22", "3389", "*"]):
+                                port = True
+                            if dest_port_ranges:
+                                i = 1
+                                while dest_port_ranges:
+                                    if dest_port_ranges.value.lower() in ["22", "3389", "*"]:
+                                        port = True
+                                        break
+                                    i += 1
+                                    dest_port_ranges = self.check_required_attribute(element.attributes, [""], f"destination_port_ranges[{i}]")
```

Comment:
Here we have the same type of pattern. I was thinking of a function such as:
```
def iterate_required_attributes(attribute, name, check)`
```
where attribute is `dest_port_ranges`, name is in this case the string `"destination_port_ranges"` and check would be a lambda function with the verification in the if.